### PR TITLE
Back icon should is a button

### PR DESCRIPTION
**Pull request by chenshuai2144**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [X] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?

https://github.com/ant-design/ant-design/issues/16038

### 💡 Solution

back icon provide better accessibility

### 📝 Changelog

- Back icon will now render as a button
- Back icon 现在会渲染成一个 button

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
这样 `backIcon={<a href="">链接</a>}` 会失效，因为规范中 button 里不能放 a。

File: components/page-header/index.tsx [hunk 1]
```diff
@@ -31,17 +32,23 @@ const renderBack = (
...
+      {({ back }: { back: string }) => (
+        <div
+          className={`${prefixCls}-back-icon`}
+          onClick={e => {
+            if (onBack) {
+              onBack(e);
+            }
+          }}
+        >
+          <TransButton className={`${prefixCls}-back-icon-button`} aria-label={back}>
```
Comment (zombieJ) [inline components/page-header/index.tsx]: onClick 到这边来

File: components/page-header/index.tsx [hunk 2]
```diff
@@ -31,17 +32,24 @@ const renderBack = (
...
+    <LocaleReceiver componentName="PageHeader">
+      {({ back }: { back: string }) => (
+        <div className={`${prefixCls}-back-icon`}>
+          <TransButton
+            onClick={(e: React.MouseEvent<HTMLDivElement>) => {
+              if (onBack) {
+                onBack(e);
+              }
+            }}
+            className={`${prefixCls}-back-icon-button`}
```
Comment (zombieJ) [inline components/page-header/index.tsx]: `${prefixCls}-back-icon-button` => `${prefixCls}-back-button`

**Review by zombieJ — APPROVED** [review]

**Comment by zombieJ** [comment]
feature branch？

**Comment by chenshuai2144** [comment]
这个更新几乎无感

**Pull request merged**